specs/agents/configuration.md

```diff
@@ -10,6 +10,50 @@ Here's a list of the config options across all agents, their types, default valu
 
 They are provided as environment variables but depending on the language there might be several feasible ways to let the user tweak them. For example besides the environment variable `ELASTIC_APM_SERVER_URL`, the Node.js Agent might also allow the user to configure the server URL via a config option named `serverUrl`, while the Python Agent might also allow the user to configure it via a config option named `server_url`.
 
+### Configuration Source Precedence
+
+Configuration can be provided via a number of sources. The sources and their
+precedence are as follows. Sources higher on this list will override values
+provided by sources lower on this list:
+
+ - Central configuration
+ - Environment variables
+ - Inline configuration in code
+ - Config files
+ - Default value
+
+### Invalid Configuration Values
+
+If an invalid value for a configuration option is provided (for example:
+`breakdown_metrics="yes"` or `apiRequestTime="1h"`) then the agent MUST ignore
+the value (falling back to a config source with lower precedence) and SHOULD
+emit a log warning about the ignored value.
+
+### Configuration Value Types
+
+The following table enumerates the available configuration types across the
+agents:
+
+
+| Type | Description (if needed) |
+|------|-------------------------|
+| String   |  |
+| Integer  |  |
+| Float    |  |
+| Boolean  | Encoded as a lower-case boolean string: `"false"`, `"true"` |
+| List     | Encoded as a comma-separated string (whitespace surrounding items should be stripped): `"foo,bar,baz"` |
+| Mapping  | Encoded as a string, with `"key=value"` pairs separated by commas (whitespace surrounding items should be stripped): `"foo=bar,baz=foo"` |
+| Duration | String with duration encoded using unit suffixes (`ms` for millisecond, `s` for second, `m` for minute). |
```

Comment:
If we list `Duration` as a type, we should also list `Size` (or `Bytesize`, or something else entirely, I'm not good with naming)

See e.g. https://www.elastic.co/guide/en/apm/agent/python/current/configuration.html#config-format-size

Reply:
Thanks! Definitely an oversight.